src/physics/shortwave_radiation.jl

```diff
@@ -86,4 +86,132 @@ function shortwave_radiation!(
+    rh_clp::Float64       # RH'_cl,     ---------------------''-----------------------
+    q_cl::Float64         # Q_cl,       ---------------------''-----------------------
+    wp_cl::Float64        # w_pcl   weighting of precipitation term (Eq. 28).
+    pmax_cl::Float64      # p_maxcl (mm/day) 
+    # optics
+    alb_cl::Float64       # A_cl (cloud albedo, visible) (Eq. 33).
+end
+
+"""
+    OneBandShortwave(; 
+        rh_cl=0.30,          # RH threshold for cloud cover = 0 (-)
+        rh_clp=1.00,         # RH threshold for cloud cover = 1 (-)
+        q_cl=0.20,           # Absolute humidity threshold for cloud cover (g kg⁻¹)
+        wp_cl=0.20,          # Weight for √precip term (-) at 1 mm day⁻¹
+        pmax_cl=10.0,        # Cap on precip contributing to cloud cover (mm day⁻¹)
+        alb_cl=0.43,         # Cloud albedo for visible band at CLC=1 (-)
+    )
+
+Defaults mirror SPEEDYs shortwave constants. 
+Notes:
+- `abs_wv_vis` expects q in g kg⁻¹. If your model uses kg kg⁻¹, internally scale by 1000 (or rescale the coefficient).
+- `abs_dry` is calibrated per layer mass of Δp = 10⁵ Pa (SPEEDY convention). Match your Beer-Lambert exponent accordingly.
+- If you dont implement Eq. 34 stratocumulus, you can ignore `alb_cls`, `cls_*`, `clf`, `gses*`.
+"""
+OneBandShortwave() = OneBandShortwave(
+    0.30, 1.00, 0.20,   # rh_cl (-), rh_clp (-), q_cl (g kg⁻¹)
+    0.20, 10.0,         # wp_cl (-), pmax_cl (mm day⁻¹)
+    0.43                # alb_cl (-)
+)
+
+get_nbands(::OneBandShortwave) = 1
+initialize!(::OneBandShortwave, ::PrimitiveEquation) = nothing
+
+
+function shortwave_radiation!(
+    column::ColumnVariables,
+    scheme::OneBandShortwave,
+    model::PrimitiveEquation,
+)
+    # ---- pull only fields that exist on ColumnVariables ----
+    (; nlayers, cos_zenith, land_fraction,
+       humid, sat_humid, albedo_ocean, albedo_land,
+       optical_depth_shortwave, flux_temp_downward, flux_temp_upward,
+       cloud_top, rain_rate_convection, rain_rate_large_scale) = column
+
+    # ---- aliases / guards ----
+    μ   = max(cos_zenith, 1f-3)                         # avoid divide-by-zero
+    qv  = humid
+    qsat = sat_humid
+    T   = eltype(qv)
+    τk = @view optical_depth_shortwave[1:nlayers, 1]
+
+    # ---- derived diagnostics ----
+    RH = @. clamp(qv / max(qsat, eps(T)), 0, 1) # clamp relative humidity between 0 and 1
+    k_cl = (1 <= cloud_top <= nlayers) ? cloud_top : max(nlayers - 1, 1) # cloud top layer index
+    RH_cl  = scheme.rh_cl 
+    RH_clp = scheme.rh_clp
+    r      = clamp((RH[min(k_cl, nlayers)] - RH_cl) / max(RH_clp - RH_cl, T(1e-6)), zero(T), one(T))
+
+    P_mmday = T(86.4) * (rain_rate_convection + rain_rate_large_scale)
+    P_mmday = min(scheme.pmax_cl, P_mmday)
+    CLC     = min(one(T), scheme.wp_cl * sqrt(P_mmday) + r^2)
+
+    # ---- transmissivity per layer: τ_layer = exp(- τk / μ) ----
+    τ = similar(qv)
+    @inbounds @simd for k in 1:nlayers
+        τ[k] = exp(-τk[k] / μ)
```

Comment:
where is τk[k] actually being written into? And why do we divide by the cos zenith here? That should be the top flux boundary condition